onadata/apps/api/tests/viewsets/test_xform_viewset.py

```diff
@@ -3110,24 +3119,28 @@ def test_csv_export_with_win_excel_utf8(self):
 
             content = get_response_content(response)
 
-            expected_content = (
-                '\xef\xbb\xbfage,\xef\xbb\xbfname,\xef\xbb\xbfmeta/instanceID,'
-                '\xef\xbb\xbf_id,'
-                '\xef\xbb\xbf_uuid,\xef\xbb\xbf_submission_time,'
-                '\xef\xbb\xbf_tags,\xef\xbb\xbf_notes,\xef\xbb\xbf_version,'
-                '\xef\xbb\xbf_duration,\xef\xbb\xbf_submitted_by,'
-                '\xef\xbb\xbf_total_media,\xef\xbb\xbf_media_count,'
-                '\xef\xbb\xbf_media_all_received\n'
-                '\xef\xbb\xbf#age,,,,,,,,,,,,,\n29,\xef\xbb\xbfLionel Messi,'
-                '\xef\xbb\xbfuuid:74ee8b73-48aa-4ced-9072-862f93d49c16,'
-                '%s,'
-                '\xef\xbb\xbf74ee8b73-48aa-4ced-9072-862f93d49c16,\xef\xbb\xbf'
-                '2013-02-18T15:54:01,\xef\xbb\xbf,\xef\xbb\xbf,'
-                '\xef\xbb\xbf201604121155,\xef\xbb\xbf,\xef\xbb\xbfbob,0,0,'
-                'True\n' % data_id
-            )
-
-            self.assertEqual(expected_content, content)
+            expected_content_py2 = (
+                '\ufeffage,\ufeffname,\ufeffmeta/instanceID,\ufeff_id,'
+                '\ufeff_uuid,\ufeff_submission_time,\ufeff_tags,\ufeff_notes,'
+                '\ufeff_version,\ufeff_duration,\ufeff_submitted_by,'
+                '\ufeff_total_media,\ufeff_media_count,'
+                '\ufeff_media_all_received\n\ufeff#age,,,,,,,,,,,,,\n29,'
+                '\ufeffLionel Messi,'
+                '\ufeffuuid:74ee8b73-48aa-4ced-9072-862f93d49c16,%s,'
+                '\ufeff74ee8b73-48aa-4ced-9072-862f93d49c16,'
+                '\ufeff2013-02-18T15:54:01,\ufeff,\ufeff,\ufeff201604121155,'
+                '\ufeff,\ufeffbob,0,0,True\n' % data_id)
+            expected_content_py3 = (
+                '\ufeffage,name,meta/instanceID,_id,_uuid,_submission_time,'
+                '_tags,_notes,_version,_duration,_submitted_by,_total_media,'
+                '_media_count,_media_all_received\n\ufeff#age,,,,,,,,,,,,,\n'
+                '\ufeff29,Lionel Messi,'
+                'uuid:74ee8b73-48aa-4ced-9072-862f93d49c16,%s,'
+                '74ee8b73-48aa-4ced-9072-862f93d49c16,2013-02-18T15:54:01,,,'
+                '201604121155,,bob,0,0,True\n' % data_id)
+
+            self.assertIn(content, [expected_content_py2,
```

Comment:
I can't tell if the content produced by py3 is correct or not. Can someone with a windows machine, or who knows in more detail if this output is correct look into this. Also note, that we can pass `decode=False` to https://github.com/onaio/onadata/pull/1295/files#diff-c9fc0b9ffbf3368f5dd3c5c05dc98840R3120 and then decode not at all to get the original test py2 content. We could also decode `utf-8-sig`, which is how this file was originally coded.

None of the decodings leads to equivalent content on py2 and py3. I believe on the encode side (in `csv_builder` and `export_builder`) we have equivalent code for the two py versions, but I'm a bit concerned since I have noticed that `open(..., encoding='utf-8')` appears to not always function equivalently, despite http://python-future.org/compatible_idioms.html#file-io-with-open